table/sst_file_writer.cc

```diff
@@ -462,6 +476,32 @@ Status SstFileWriter::Finish(ExternalSstFileInfo* file_info) {
 
   if (file_info != nullptr) {
     *file_info = r->file_info;
+    // Remove user-defined timestamps from external file metadata too when they
+    // should not be persisted.
+    if (r->strip_timestamp) {
```

Comment:
We could consider adding some assertions here to the effect of "if smallest key is empty, so is largest key" (and similarly for the range deletion keys).